src/main/java/tasklist/TaskList.java

```diff
@@ -0,0 +1,139 @@
+package tasklist;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import exception.ErrorAccessingFile;
+import exception.NonExistentTaskNumberException;
+import storage.StorageFile;
+
+/**
+ * Encapsulates a task list storing the users input tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> list = new ArrayList<>();
+    private StorageFile listFile;
+
+    public TaskList(StorageFile listFile) {
+        this.listFile = listFile;
+    }
+
+    /**
+     * Scans tasks from storage to list.
+     *
+     * @param task Task to be scanned into list.
+     */
+    public void scanExistingTaskToList(Task task) {
+        this.list.add(task);
+    }
+
+    /**
+     * Adds task to a list.
+     *
+     * @param task Task be added to the list
+     */
+    public void addTaskToList(Task task) throws ErrorAccessingFile {
+        try {
+            this.listFile.add(task.toString());
+            this.list.add(task);
+        } catch (IOException e) {
+            throw new ErrorAccessingFile("add");
+        }
+    }
+
+    /**
+     * Deletes task from a list.
+     *
+     * @param taskNumber Number of the task to be removed from the list.
+     */
+    public Task deleteTaskFromList(int taskNumber) throws NonExistentTaskNumberException, ErrorAccessingFile {
+        try {
+            validateTaskNumberExists(taskNumber);
+
+            Task task = this.getTaskByTaskNumber(taskNumber);
+            this.list.remove(taskNumber - 1);
+            this.listFile.rewriteFile(this.list);
+
+            return task;
+        } catch (IOException e) {
+            throw new ErrorAccessingFile("delete");
+        }
+    }
+
+    /**
+     * Marks task in the list as done, finding the task by its task number.
+     *
+     * @param taskNumber Task number of the task in the list, starting from 1.
+     * @return Task.
```

Comment:
To be consistent with your other documentations, consider making this into code using markdown ``